### ISSUE TITLE
Add semantic versioning support

### ISSUE THREAD

**Issue by MuratUrsavas** [NebraLtd/helium-miner-software#351]
It would be better to switch to semantic versioning, where application version is controlled automatically via GitHub.

One point is missing on my side, though. The documentation clearly explains how to bump the major and minor version numbers, but it's not clear how to bump maintenance and prerelease numbers.

**Comment by vpetersson** [comment]
This is a fair bit complicated than it seems. We discussed this in today's stand-up and decided that we need to scope this further in grooming but here are the cliffnotes:

* We should adopt full semantic versioning for the Nebra software suite
* This should follow standard semantic versioning (major, minor, patch)
* This should be fully detached from the Helium Miner GA versions
  * The GA version will be *part* of the release and it's TBD how this will affect the version bump
* We need to have clear guidelines on what constitutes a major/minor/patch release
* Each release must have clear release notes

Semi-relevant ticket:
* https://github.com/NebraLtd/hm-diag/issues/290

**Comment by shawaj** [comment]
@vpetersson @marvinmarnold are we still thinking to include semantic versioning at some point? or is the nebra software version hash thing sufficient for now?

**Comment by vpetersson** [comment]
Yeah i still want to add proper semantic versioning for this as it will be a lot easier to generate change logs for.
